components/SnippetDetails.js

```diff
@@ -0,0 +1,36 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+function SnippetDetails() {
+  const router = useRouter();
+
+  const { id } = router.query;
+
+  const { data, isLoading, error } = useSWR(`/api/snippets/${id}`);
+
+  if (error) return <div>failed to load Details View 🥺</div>;
+  if (isLoading) {
+    return <h1>Loading Details View 🤓</h1>;
+  }
+  if (!data) {
+    return <div>cant find the data yet</div>;
+  }
```

Comment:
as soon as `error` and `isLoading` are false, there should be data. So I don't think you will need these lines